src/neuroconv/tools/roiextractors/roiextractors.py

```diff
@@ -672,15 +680,56 @@ def image_mask_iterator():
                     description="1 if ROI was rejected or 0 if accepted as a cell during segmentation operation",
                 ),
             ],
-            id=roi_ids,
         )
+        if mask_type != "pixel" and mask_type != "voxel":  # roi_ids will not be written row-wise
+            plane_segmentation_kwargs.update(id=roi_ids)
+
         plane_segmentation = PlaneSegmentation(**plane_segmentation_kwargs)
+
         if include_roi_centroids:
             tranpose_roi_location_axes = (1, 0) if len(segmentation_extractor.get_image_size()) == 2 else (1, 0, 2)
             roi_locations = segmentation_extractor.get_roi_locations()[tranpose_roi_location_axes, :].T
             plane_segmentation.add_column(
                 name="ROICentroids", description="The x, y, (z) centroids of each ROI.", data=roi_locations
             )
+
+        if mask_type is not "None" and mask_type == "image":
+            plane_segmentation.add_column(
```

Comment:
First difficulty - the JSON schema validator complained if I set the upstream annotation to `Optional[str]` and set the value to `None` (not a 'recognized' JSON type).

Best fix would probably be to allow the schema parser to understand the Python `None` type.

Reply:
Mmmmm so hacky, I really don't like that lol. But I suppose it would work

Reply:
Yep, not sure what's better, we are choosing between two ugly options here I think. I agree that the best global fix is what you describe. Fixing the parser.

Reply:
Trying to fix now then

Reply:
Fixed. `Optional[...]` wrappers now work in any signature. As long as it's also not containing a `Union` of other types, which could indicate conflicts.